React/Views/RCTNavItem.m

```diff
@@ -11,10 +11,23 @@
 
 @implementation RCTNavItem
 
+@synthesize titleImageView = _titleImageView;
 @synthesize backButtonItem = _backButtonItem;
 @synthesize leftButtonItem = _leftButtonItem;
 @synthesize rightButtonItem = _rightButtonItem;
 
+- (UIImageView *)titleImageView
+{
+  if (!_titleImageView) {
+    if (_titleImage) {
```

Comment:
This will not be properly invalidated when you update the `titleImage` property.